libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -801,6 +802,13 @@ bool AP_Arming::system_checks(bool report)
             check_failed(ARMING_CHECK_SYSTEM, report, "Scripting out of memory");
             return false;
         }
+#endif
+#if HAL_ADSB_ENABLED
+        AP_ADSB *adsb = AP::ADSB();
+        if ((adsb != nullptr) && adsb->init_failed()) {
```

Comment:
```suggestion
        if ((adsb == nullptr) || adsb->init_failed()) {
```

Better to be over-cautious than under-cautious.

Reply:
Your suggestion backfired, the nullptr check was true in rover. I changed the logic to match the other checks